api/src/main/java/org/hawkular/apm/api/services/AnalyticsService.java

```diff
@@ -190,50 +190,51 @@
                                 Criteria criteria, boolean asTree);
 
     /**
+     * This method returns the list of host names where activities were executed, subject to the supplied
+     * criteria.
+     *
+     * @param tenantId The tenant id
+     * @param criteria The criteria
+     * @return The list of host names
+     */
+    List<String> getHostNames(String tenantId, Criteria criteria);
```

Comment:
This seems unrelated to the "theme" of this PR. I'm much worse than this when it comes to keeping focus of the PRs, so I actually don't mind :wink: Just pointing out it is discouraged.